### PR TITLE
framework updates

### PR DESCRIPTION
**Pull request by aatchison14**
removed .NET Core 3.1
added .NET 8

type of change:
breaking change (fix or feature that could cause existing functionality to not work as expected)

https://git.rockfin.com/Rocket-Libraries/Planning/issues/881

### CODE REVIEW

File: RockLib.Compression/CHANGELOG.md
Comment (JasonBock) [inline RockLib.Compression/CHANGELOG.md:1]: I think we'll want to push a `3.0.0-alpha.1` first.

File: RockLib.Compression/RockLib.Compression.csproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <PublishRepositoryUrl>True</PublishRepositoryUrl>
     <EmbedUntrackedSources>True</EmbedUntrackedSources>
     <DebugType>Embedded</DebugType>
+    <TargetFrameworks>net6.0;net8.0;net48</TargetFrameworks>
```
Comment (JasonBock) [inline RockLib.Compression/RockLib.Compression.csproj:20]: I think this should be updated in `Directory.Build.props`....while you're in there, you'll want to change some of the `PropertyGroup` conditions.

**Review by JasonBock — CHANGES_REQUESTED** [review]

File: RockLib.Compression/CHANGELOG.md
Comment (aatchison14) [reply RockLib.Compression/CHANGELOG.md:1]: doh!

File: RockLib.Compression/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 3.0.0-alpha.1 - 2024-01-03
+
+#### Changed
+
+- Supported targets: net48, net6.0, and net8.0.
```
Comment (kristycurrier) [inline RockLib.Compression/CHANGELOG.md]: I would spell out that support for .NET Core 3.1 was removed.

**Review by JasonBock — APPROVED** [review]

**Review by kristycurrier — APPROVED** [review]

**Review by kristycurrier — CHANGES_REQUESTED** [review]
Just looked, are there any GitHub actions setup for this repo?

**Comment by aatchison14** [comment]
> Just looked, are there any GitHub actions setup for this repo?

oh, no there are not. I'll get that in place

**Comment by aatchison14** [comment]
> Just looked, are there any GitHub actions setup for this repo?

all done

**Review by kristycurrier — APPROVED** [review]

**Pull request merged**